Core/include/Acts/Propagator/Navigator.hpp

```diff
@@ -248,937 +259,483 @@ class Navigator {
+    if (state.startSurface != nullptr &&
+        state.startSurface->associatedLayer() != nullptr) {
       ACTS_VERBOSE(
           volInfo(state)
           << "Fast start initialization through association from Surface.");
 
-      // assign the current layer and volume by association
-      state.navigation.startLayer =
-          state.navigation.startSurface->associatedLayer();
-      state.navigation.currentLayer = state.navigation.startLayer;
-
-      state.navigation.startVolume =
-          state.navigation.startLayer->trackingVolume();
-      state.navigation.currentVolume = state.navigation.startVolume;
-    } else if (state.navigation.startVolume != nullptr) {
+      state.startLayer = state.startSurface->associatedLayer();
+      state.startVolume = state.startLayer->trackingVolume();
+    } else if (state.startVolume != nullptr) {
       ACTS_VERBOSE(
           volInfo(state)
           << "Fast start initialization through association from Volume.");
 
-      state.navigation.currentVolume = state.navigation.startVolume;
-
-      state.navigation.startLayer =
-          state.navigation.startVolume->associatedLayer(
-              state.geoContext, stepper.position(state.stepping));
-      state.navigation.currentLayer = state.navigation.startLayer;
+      state.startLayer = state.startVolume->associatedLayer(
+          state.options.geoContext, position);
     } else {
       ACTS_VERBOSE(volInfo(state)
                    << "Slow start initialization through search.");
-      // current volume and layer search through global search
       ACTS_VERBOSE(volInfo(state)
-                   << "Starting from position "
-                   << toString(stepper.position(state.stepping))
-                   << " and direction "
-                   << toString(stepper.direction(state.stepping)));
-
-      state.navigation.startVolume =
-          m_cfg.trackingGeometry->lowestTrackingVolume(
-              state.geoContext, stepper.position(state.stepping));
-      state.navigation.currentVolume = state.navigation.startVolume;
-
-      if (state.navigation.startVolume != nullptr) {
-        state.navigation.startLayer =
-            state.navigation.startVolume->associatedLayer(
-                state.geoContext, stepper.position(state.stepping));
-        state.navigation.currentLayer = state.navigation.startLayer;
-        ACTS_VERBOSE(volInfo(state) << "Start volume resolved.");
+                   << "Starting from position " << toString(position)
+                   << " and direction " << toString(direction));
+
+      // current volume and layer search through global search
+      state.startVolume = m_cfg.trackingGeometry->lowestTrackingVolume(
+          state.options.geoContext, position);
+
+      if (state.startVolume != nullptr) {
+        state.startLayer = state.startVolume->associatedLayer(
+            state.options.geoContext, position);
       } else {
         ACTS_VERBOSE(volInfo(state)
```

Comment:
Good point. I guess this only happens if the tracking geometry is broken or somebody starts the propagation outside. The numerical integration would still be valid as long as there is a magnetic field but we will never be able to enter the tracking geometry. So I guess it is an error? Otherwise we silently fall back to void navigation.